### PR TITLE
[FMv0.6]--Update docs about auth support in Mesh Worker Service

### PR DESCRIPTION
**Pull request by jiangpengcheng**
### Motivation

Function Mesh Worker Service v0.6.0 support using separate auth for each user(only support OAuth2)([code PR #357](https://github.com/streamnative/function-mesh-worker-service/pull/357)), therefore add docs accordingly.

### Modification

- Update the Function Mesh Worker Service overview docs

### CODE REVIEW

File: docs/function-mesh-worker/function-mesh-worker-overview.md [hunk 1]
```diff
@@ -16,6 +16,34 @@ The following figure illustrates how Function Mesh Worker service works with Pul
 
 ![Function Mesh Workflow](./../assets/function-mesh-workflow.png)
 
+## Authentication Support
+
+Before v0.6.0, the Function Mesh Worker Service uses a super admin Pulsar account reading from the config file for Function Mesh operators, which is not secure, so start from v0.6.0, there is a break change to use separate account for each user.
+
+**Currently it only support OAuth2 providers, other providers will be supported as soon as possible.**
```
Comment (Huanli-Meng) [inline docs/function-mesh-worker/function-mesh-worker-overview.md:23]:
```suggestion
> **Note**
>
> Currently, only OAuth2 authentication is supported.
```

File: docs/function-mesh-worker/function-mesh-worker-overview.md [hunk 2]
```diff
@@ -16,6 +16,34 @@ The following figure illustrates how Function Mesh Worker service works with Pul
...
+
+> NOTE: if you are using Pulsar cluster with other authentication providers, the Function Mesh Worker Service will not work
```
Comment (Huanli-Meng) [inline docs/function-mesh-worker/function-mesh-worker-overview.md:25]: we can remove this note as we have expressed similar meaning in the previous sentence.

File: docs/function-mesh-worker/function-mesh-worker-overview.md [hunk 3]
```diff
@@ -16,6 +16,34 @@ The following figure illustrates how Function Mesh Worker service works with Pul
...
+
+### OAuth2
+
+For each client of OAuth2, there should be a Kubernetes Secret created in advance in the same cluster of Function Mesh operators, its format is like below:
```
Comment (Huanli-Meng) [inline docs/function-mesh-worker/function-mesh-worker-overview.md:29]:
```suggestion
For clients that use OAuth2 authentication, you need to create a Kubernetes Secret in advance in the same cluster where Function Mesh operators work. The Kubernetes Secret is something like the below:
```

File: docs/function-mesh-worker/function-mesh-worker-overview.md [hunk 4]
```diff
@@ -16,6 +16,34 @@ The following figure illustrates how Function Mesh Worker service works with Pul
...
+      "client_secret": "xxxx",
+      "issuer_url": "xxx"
+    }
+  clientAuthenticationParameters: {"privateKey":"file:///${SOME-PATH-OF-FILE}","issuerUrl":"${OAUTH-ISSUER-URL}","audience":"${OAUTH-AUDIENCE}","scope":"${OAUTH-SCOPE}"}
+kind: Secret
+metadata:
+  name: ${OAUTH-CLIENT-NAME}
+type: Opaque
+```
+
```
Comment (Huanli-Meng) [inline docs/function-mesh-worker/function-mesh-worker-overview.md:46]: For details about how to configure OAuth2 authentications for Pulsar Functions, Sources, and Sinks, see [authentication](/functions/function-crd.md#authentication).

File: docs/function-mesh-worker/function-mesh-worker-overview.md [hunk 5]
```diff
@@ -28,4 +56,4 @@ This table lists the version mapping relationships between Function Mesh and Fun
 | v0.2.0| <br />- v2.10.0.5-v2.10.0.6 <br />- v2.9.2.18-v2.9.2.22 <br />- v2.8.3.4 |
 | v0.3.0| <br />- v2.10.0.7 <br />- v2.9.2.23+ <br />- v2.8.3.5+ |
 | v0.4.0| <br />- v2.10.1.4+ <br />- v2.9.3.3+ <br />- v2.8.3.6+ |
-| v0.5.0| <br />- v2.10.1.7+ <br />- v2.9.3.5+ <br />- v2.8.4.1+ |
\ No newline at end of file
+| v0.5.0| <br />- v2.10.1.7+ <br />- v2.9.3.5+ <br />- v2.8.4.1+ |
```
Comment (Huanli-Meng) [inline docs/function-mesh-worker/function-mesh-worker-overview.md:59]: Confirmed with Rui, For FM v0.5, the FM worker service versions should be the following. Could you please also help update it? Thanks.
Function Mesh 0.5.0:
v2.10.1.7 ～ 2.10.1.8
v2.9.3.6 ～2.9.3.8
v2.8.4.1

**Review by Huanli-Meng — COMMENTED** [review]
Leave some comments. PTAL

File: docs/function-mesh-worker/function-mesh-worker-overview.md (hunk 4 above)
Comment (jiangpengcheng) [reply docs/function-mesh-worker/function-mesh-worker-overview.md:46]: 如果使用了Function Mesh Worker Service的话，用户不用去手动配置 authentication 了

File: docs/function-mesh-worker/function-mesh-worker-overview.md (hunk 5 above)
Comment (jiangpengcheng) [reply docs/function-mesh-worker/function-mesh-worker-overview.md:61]: ok

**Review by Huanli-Meng — APPROVED** [review]
LGTM

**Comment by Huanli-Meng** [comment]
@nlu90 @freeznet could any of you help review the PR? Thanks.

**Review by nlu90 — APPROVED** [review]

**Pull request merged**